packages/mermaid/src/diagrams/state/stateDb.js

```diff
@@ -368,10 +374,10 @@ function endTypeIfNeeded(id = '', type = DEFAULT_STATE_TYPE) {
  * @param relationTitle
  */
 export function addRelationObjs(item1, item2, relationTitle) {
-  let id1 = startIdIfNeeded(item1.id);
-  let type1 = startTypeIfNeeded(item1.id, item1.type);
-  let id2 = startIdIfNeeded(item2.id);
-  let type2 = startTypeIfNeeded(item2.id, item2.type);
+  let id1 = startIdIfNeeded(item1.id.trim());
+  let type1 = startTypeIfNeeded(item1.id.trim(), item1.type);
+  let id2 = startIdIfNeeded(item2.id.trim());
+  let type2 = startTypeIfNeeded(item2.id.trim(), item2.type);
```

Comment:
Trimming - we could definitely trim() it inside startIdIfNeeded().  I was keeping that function small (trying to keep # responsibilities small), but the line is fuzzy.
I wanted to make sure to trim() everything before storing it (descriptions, ids, etc.) because I've found that the parsers are consistent across diagrams.
I did think about "where should trim() happen so it's consistent but not done before it's needed" and didn't really come to a conclusion (obv LOL).

I wanted to stay away from doing much with the parser with this PR; wanted to try to keep the scope kind of small to try to get this thru a little faster.
But I can definitely look in to that if needed.